### PR TITLE
Feat(grid): add radio button selection

### PR DESCRIPTION
**Pull request by CatalinB7**
TODO: 
- [x] add tests

### CODE REVIEW

File: projects/angular/components/ui-grid/src/ui-grid.component.html [hunk 1]
```diff
@@ -347,6 +348,11 @@
                               tabindex="0">
                 </mat-checkbox>
             </div>
+            <div *ngIf="singleSelectable"
+                 class="ui-grid-mobile-feature-container ui-grid-mobile-refresh-container">
+                <mat-radio-button></mat-radio-button>
```
[the remark this answers is not in the document]
Comment (CatalinB7) [reply projects/angular/components/ui-grid/src/ui-grid.component.html]: it should 👍

File: projects/angular/components/ui-grid/src/ui-grid.component.html [hunk 2]
```diff
@@ -347,6 +348,11 @@
                               tabindex="0">
                 </mat-checkbox>
             </div>
+            <div *ngIf="singleSelectable"
+                 class="ui-grid-mobile-feature-container ui-grid-mobile-refresh-container">
+                <mat-radio-button (change)="rowSelected(row)"></mat-radio-button>
```
Comment (anbalase) [inline projects/angular/components/ui-grid/src/ui-grid.component.html:353]: see checkbox state logic binding to selection manager.
see disabled state
see tooltips

add tests for all of the above

**Review by anbalase — CHANGES_REQUESTED** [review]

File: projects/angular/components/ui-grid/src/ui-grid.component.html [hunk 3]
```diff
@@ -332,21 +333,30 @@
...
-                              (change)="handleSelection(index, row)"
-                              [checked]="selectionManager.isSelected(row)"
-                              [indeterminate]="selectionManager.isIndeterminate(row)"
-                              [matTooltip]="disabledReason || checkboxTooltip(row)"
-                              [aria-label]="disabledReason || checkboxTooltip(row)"
-                              [disabled]="!!disabledReason"
-                              tabindex="0">
-                </mat-checkbox>
-            </div>
+            <ng-container *ngLet="disableSelectionByEntry(row) as disabledReason">
```
Comment (anbalase) [inline projects/angular/components/ui-grid/src/ui-grid.component.html:336]: can I have both on?

Comment (CatalinB7) [reply projects/angular/components/ui-grid/src/ui-grid.component.html:336]: nope, solved

**Review by caleniuc — DISMISSED** [review]

**Review by alexdudescu — APPROVED** [review]

**Pull request merged**